helix-gateway/src/main/java/org/apache/helix/gateway/service/GatewayServiceManager.java

```diff
@@ -127,8 +132,13 @@ public void run() {
     }
   }
 
-  public HelixGatewayServiceChannel getHelixGatewayServiceProcessor() {
-    return _gatewayServiceChannel;
+
+  public void startService() throws IOException {
+    _gatewayServiceChannel.start();
+  }
+
+  public void stopService() {
+    _gatewayServiceChannel.stop();
```

Comment:
You just stop the channel. processor needs to be stopped as well. And clean the map.